mmdet/datasets/coco.py

```diff
@@ -109,11 +109,17 @@ def _parse_ann_info(self, img_info, ann_info):
         gt_labels = []
         gt_bboxes_ignore = []
         gt_masks_ann = []
-
         for i, ann in enumerate(ann_info):
             if ann.get('ignore', False):
                 continue
             x1, y1, w, h = ann['bbox']
+            x2 = x1 + w
+            y2 = y1 + h
+            if ((x1 < 0 or x1 >= img_info['width'] or y1 < 0
+                 or y1 >= img_info['height'])
+                    and (x2 < 0 or x2 >= img_info['width'] or y2 < 0
+                         or y2 >= img_info['height'])):
+                continue
```

Comment:
I have some questions about the condition statement,
1. The conditions seem are redundant. For example if `x1 >= img_info['width']` then x2 must `>= img_info['width']`, so we only need to check `x2 >= img_info['width']`.
2. Why you use `and` between the first condition and the second condition,  in my opinion, if one of two conditions satisfy then the image is out-of-frame.

Reply:
1. Yeah, the conditions are redundant. But it is more readable. The first condition checks if the left-top point is out of image, and the second one checks the bottom-right point.
2. I mean to filter boxes that are fully outside the image (frame), but preserve those partially outside the image. In another word, when the boxes do not intersect with the image, then they are filtered. 

I think I should check the intersection instead. Current conditions are True (both corner points are out of frame) when the image is inside a box, and the box is filtered. But the box should be preserved since it intersects with the image.